[CDP] Update pagination text display logic for copays

josephrlee opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Refactored Balances component to use fragments instead of divs and article since the component was already wrapped in an article in the parent page
- Conditionally rendered pagination text p tag to only display when text is available.
- Modified SummaryPage to hide pagination text when total items are 10 or fewer, improving UI clarity.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#125441
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#40151


## Testing done

- Tested locally

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| Before | After |
| ------ | ----- |
| <img width="708" height="672" alt="Screenshot 2025-12-10 at 3 16 54 PM" src="https://github.com/user-attachments/assets/209a654a-e105-4985-bfe8-84b2b37844f5" /> | <img width="760" height="635" alt="Screenshot 2025-12-10 at 3 16 10 PM" src="https://github.com/user-attachments/assets/53f070ee-18e5-4dc6-b3a1-60019668d491" /> |

## What areas of the site does it impact?

Combined debt portal

## Acceptance criteria

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [X] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

josephrlee committed:
Update Balances and pagination text display logic

- Refactored Balances component to use fragments instead of divs and article since the component was already wrapped in an article in the parent page
- Conditionally rendered pagination text p tag to only display when text is available.
- Modified SummaryPage to hide pagination text when total items are 10 or fewer, improving UI clarity.

OptionSelect reviewed (commented):
Lets make that 10 you have in there use the MAX_ROWS variable we have up top! Otherwise great to go!

josephrlee committed:
Use MAX_ROWS constant for pagination check

Replaces the hardcoded value of 10 with the MAX_ROWS constant when determining whether to display pagination text. This improves maintainability and ensures consistency if the maximum number of rows changes.

OptionSelect reviewed (approved):
LGTM! 🚀
